src/App.tsx

```diff
@@ -37,15 +38,39 @@ const App = () => {
 		// what Metamask injects as window.ethereum into each page
 		ethProvider = new ethers.providers.Web3Provider(window.ethereum);
 	} catch (e) {
-		return (
-			<EthError />
-		);
+		return (<ErrorMessage>
+			<>
+				<h1>Couldn&apos;t connect to Rialto Ethereum node</h1>
+				<h3>
+							Make sure to install & enable
+					<img
+						src='https://avatars0.githubusercontent.com/u/11744586?s=280&v=4'
+					/>
+							Metamask extension
+				</h3>
+			</>
+		</ErrorMessage>);
 	}
 
 	// The Metamask plugin also allows signing transactions to
 	// send ether and pay to change state within the blockchain.
 	// For this, we need the account signer...
 	window.ethereum.enable();
+	window.ethereum.on('chainChanged', () => {
+		// Handle the new chain.
+		// Correctly handling chain changes can be complicated.
+		// We recommend reloading the page unless you have a very good reason not to.
+		window.location.reload();
+	});
+
+	if (Number(window.ethereum.networkVersion) !== EXPECTED_NETWORK_ID){
+		return (<ErrorMessage>
+			<>
+				<h1><Icon name='warning circle'/>Unexpected network</h1>
+				<h3>Please connect Metamask to Rialto (using {WS_PROVIDER})</h3>
```

Comment:
This should rather be `http://rpc.rialto.brucke.link:8545` not the `wss` stuff (which is Substrate node)

Reply:
ah thanks